bftengine/src/bftengine/messages/PrePrepareMsg.cpp

```diff
@@ -156,7 +156,7 @@ PrePrepareMsg::PrePrepareMsg(ReplicaId sender,
                   spanContext.data().size(),
                   (((size + sizeof(Header) + batchCid.size()) < maxMessageSize<PrePrepareMsg>())
                        ? (size + sizeof(Header) + batchCid.size())
-                       : maxMessageSize<PrePrepareMsg>() - spanContext.data().size())) {
+                       : maxMessageSize<PrePrepareMsg>() - MessageBase::SPAN_CONTEXT_MAX_SIZE)) {
```

Comment:
Thanks for offline clarification about changing this logic.
There are multiple conditions here and some calculation being done, If possible can we add a unit test in PrePrepareMsg_test.cpp to make sure that sizeof PPM doesn't go beyond max_external_msg_size?

Reply:
Thanks for test.